src/SwedbankPay.Sdk/Payments/PaymentsResource.cs

```diff
@@ -1,46 +1,26 @@
-﻿using System;
+﻿using SwedbankPay.Sdk.Payments.CardPayments;
+using SwedbankPay.Sdk.Payments.SwishPayments;
+using System;
+using System.Net.Http;
 using System.Threading.Tasks;
 
-using SwedbankPay.Sdk.Payments.Card;
-
 namespace SwedbankPay.Sdk.Payments
 {
-    internal class PaymentsResource : ResourceBase, IPaymentsResource
+    public class PaymentsResource: ResourceBase, IPaymentsResource
     {
-        public PaymentsResource(SwedbankPayHttpClient swedbankPayHttpClient)
-            : base(swedbankPayHttpClient)
-        {
-        }
-
-
-        public async Task<Payment> GetCreditCardPayment(Uri id, PaymentExpand paymentExpand = PaymentExpand.None)
-        {
-            if (id == null)
-                throw new ArgumentNullException(nameof(id));
-
-            return await Payment.Get(id, this.swedbankPayHttpClient, GetExpandQueryString(paymentExpand));
-        }
-
-
-        public async Task<Payment> CreateCreditCardPayment(PaymentRequest paymentRequest, PaymentExpand paymentExpand = PaymentExpand.None)
-        {
-            return await Payment.Create(paymentRequest, this.swedbankPayHttpClient, GetExpandQueryString(paymentExpand));
-        }
-
-
-        public async Task<Swish.Payment> GetSwishPayment(Uri id, PaymentExpand paymentExpand = PaymentExpand.None)
+        public ICardPaymentsResource CardPayments { get; private set ; }
+        public ISwishPaymentsResource SwishPayments { get; private set; }
+        
+        public PaymentsResource(HttpClient httpClient) : base(httpClient)
```

Comment:
```suggestion
        public PaymentsResource(HttpClient httpClient)
            : this(httpClient, new CardPaymentsResource(httpClient), new SwishPaymentsResource(httpClient))
```

Reply:
Doing this will give me `Error	CS7036	There is no argument given that corresponds to the required formal parameter 'httpClient' of 'ResourceBase.ResourceBase(HttpClient)'`